### ISSUE TITLE
[Accessibility]: Provide more context for filter box in ref panel

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36573]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

In the ref panel, the "Type to filter by filename" text field announces itself, but it doesn't provide any context on how many files there are, or how many matches there are after you've typed something.

This means that a screen reader user needs to scroll down and figure out how many matches there were.

### Expected behavior

It would be nice to provide some context, such as the number of files and the number of matches as you type. This should be announced when focus switches to the text field, as well as after typing some text (perhaps with some latency, so it doesn't overlap with the typed keys being read out).

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
